### PR TITLE
Check in Manifest for docs, update build scripts and README

### PR DESCRIPTION
**Pull request by simonbyrne**
Follow on from #420.

### CODE REVIEW

**Comment by jakebolewski** [comment]
I guess we will have to update these every version bump?

**Comment by simonbyrne** [comment]
I was hoping CompatHelper would take care of that?

**Comment by simonbyrne** [comment]
Oh, you mean version bump of ClimaCore? I guess we should just have that part of the release process.

**Comment by jakebolewski** [comment]
It seems like it checks in the version of the 'dev' subpackages which I didn't realize (I thought it would just be at the current commit).  Maybe this info is ignored  though.

**Comment by simonbyrne** [comment]
Yeah, I don't think it makes a difference, but we can make sure to update these when we bump the versions.

File: examples/README.md [hunk 1]
```diff
@@ -9,17 +9,10 @@
...
-Activate the Pkg repl mode and dev the ClimaCore repo and ClimaCorePlots packages into the examples project environment:
-
-    (examples) pkg> dev . lib/ClimaCorePlots
-    (examples) pkg> instantiate
+Instantiate the examples environment (this will download and install the necessary packages)
 
-Or alternatively:
-
-    julia --project=examples -e 'using Pkg; Pkg.develop([Pkg.PackageSpec(path="."), Pkg.PackageSpec(path="lib/ClimaCorePlots")])'
+    cd ClimaCore
```
Comment (jakebolewski) [inline examples/README.md:14]: I would leave a note in here that the manifest `dev` the lib/ folder sub-projects, otherwise it could be confusing

**Review by jakebolewski — APPROVED** [review]

**Pull request merged**